### PR TITLE
[et] Prebuilding iOS libraries

### PR DESCRIPTION
**Pull request by tsapeta**
# Why

We want to make compilation faster by providing prebuilt binaries of our native iOS modules.
Followup #11169 #11183 

# How

Added `et prebuild-packages`/`et prebuild` expotools command that does a few things on given package:
- Takes the podspec with sources (`EXPO_USE_SOURCE=1`)
- Converts that podspec to [XcodeGen](https://github.com/yonaskolb/XcodeGen)'s project spec
- Uses XcodeGen to generate `.xcodeproj` file based on that project spec
- Builds the library to two frameworks — one for the device (arm64) and one for simulator (arm64 & x86_64) — both in release mode
- Puts these frameworks into the new `.xcframework` format which is a wrapper for the standard frameworks but built for different architectures
- Cleans up the temporary files such as `.xcodeproj`, project spec and generated Info.plist

In terms of the XcodeGen tool — there is no gem for it, so it seemed better to me to create a new package `@expo/xcodegen` that ships the newest version (2.18.0) of XcodeGen as a binary.

# Test Plan

I've run `et prebuild-packages` and confirmed that some packages now has `.xcframework` file in its iOS subdirectory. It contains, as expected, two flavored `.framework`s inside.
Then ran `EXPO_USE_SOURCE=0 pod install` in `ios` directory, confirmed those packages have only headers and `.xcframework` in project's file explorer and then successfully built Expo Go.

# Example

Standard succeeding use

Screen Shot 2020-12-07 at 11 22 17

Command failed due to missing `<Foundation/Foundation.h>` (`xcodebuild`'s output prettified and piped to the current process)

Screen Shot 2020-12-07 at 11 37 09

### CODE REVIEW

File: tools/expotools/src/prebuilds/Prebuilder.ts [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+
+      if (paths.length > 0) {
+        return path.join(cwd, paths[0]);
+      }
+    }
+  }
+  return null;
+}
+
+//#endregion
```
Comment (bbarthec) [inline tools/expotools/src/prebuilds/Prebuilder.ts:240]: 🤔

**Review by bbarthec — APPROVED** [review]
Looks fine to me 👍 
As always this code of yours is very easy to read 🎖️

File: tools/expotools/src/prebuilds/Prebuilder.ts (hunk 1 above)
Comment (tsapeta) [reply tools/expotools/src/prebuilds/Prebuilder.ts:240]: Ah, sorry, I've been looking for something like `#pragma mark` or `//region` in Android Studio but for VSCode. Ultimately I wanted to bring this up on the meeting.

**Review by lukmccall — APPROVED** [review]

**Comment by sapramit** [comment]
@tsapeta You just pre-built it for arm64 for devices. Why is there no support for armv7? Also, there is some possibility to add the armv7 support. Thanks

**Comment by tsapeta** [comment]
@sapramit Why do you need armv7 support? This architecture is no longer used in iOS devices for years. You can probably remove it from your supported archs in Xcode project 😉

**Comment by sapramit** [comment]
Thanks for the response. Just did the same. 😊

**Pull request merged**